src/Nethermind/Nethermind.Consensus/IPendingTransactionSelector.cs

```diff
@@ -16,11 +16,12 @@
 
 using System.Collections.Generic;
 using Nethermind.Core;
+using Nethermind.Core.Crypto;
 
 namespace Nethermind.Consensus
 {
     public interface IPendingTxSelector
     {
-        IEnumerable<Transaction> SelectTransactions(long gasLimit);
+        IEnumerable<Transaction> SelectTransactions(Keccak stateRoot, long gasLimit);
```

Comment:
stateRoot or maybe ParentHeader would make more sense from API point of view?

Reply:
yeah, very good idea